src/topology/multicodec-topology.js

```diff
@@ -55,43 +55,44 @@ class MulticodecTopology extends Topology {
 
   /**
    * Update topology.
-   * @param {Array<PeerInfo>} peerInfoIterable
+   * @param {Array<{id: PeerId, multiaddrs: Array<Multiaddr>}>} peerDataIterable
    * @returns {void}
    */
-  _updatePeers (peerInfoIterable) {
-    for (const peerInfo of peerInfoIterable) {
-      if (this.multicodecs.filter(multicodec => peerInfo.protocols.has(multicodec)).length) {
+  _updatePeers (peerDataIterable) {
+    for (const peerData of peerDataIterable) {
```

Comment:
nit: exploding the properties might tidy this function up a little.
```js
for (const { id, protocols }  of peerDataIterable) {
```

Reply:
done :)